react-datepicker/index.d.ts

```diff
@@ -25,10 +25,9 @@ declare module "react-datepicker" {
         maxDate?: {};
         minDate?: {};
         name?: string;
-        onBlur?():any;
-        onChange():any;
-        onChange(date?:any):any;
-        onFocus?():any;
+        onBlur?(handler: (e: any) => void): any;
+        onChange(handler: (date?: any, e?: any) => void): any;
```

Comment:
Callback type parameters shouldn't be marked optional. A function `() => {}` satisfies the type `(date: any, e: any) => void` just fine.